[Feat/study_group] 스터디 멤버 관련 모델 작성, 모델 패키지화, TimeStampedModel 적용

TeaCat-Develop opened the pull request:
## ✅ PR 요약
작업 요약: 모델 추가 및 패키지화

## 📄 상세 내용
- [x] 충돌 해결을 위한 recruitments 마이그레이션 파일 삭제 및 makemigrations (2팀 hotfix와 rebase)
- [x] 유저 테이블 참조 모델명 Users -> User 수정
- [x] 모델명 정리 (테이블명과 동일하게 수정)
  - StudyGroupBaseModel -> StudyGroup
  - GroupMemberBaseModel -> GroupMember
- [x] Models.model 대신 core의 TimeStampedModel 적용

## 🧪 PR Checklist
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트)

TeaCat-Develop committed:
✨ feat: GroupMemberBaseModel 모델 추가 및 패키지화

class Meta: db_table 이름 추가
GroupMemberBaseModel 작성
기존 models.py를 models 디렉토리 아래로 이동
모델을 파일 별로 나누고 __init__.py 파일을 사용해 패키지화

패키지화로 인해 순환참조이슈 발생
-> import 대신 문자열로 FK 참조하여 해결
-> ForeignKey("app_name.ModelName", on_delete=models.CASCADE)

TeaCat-Develop committed:
🐛 fix: 참조 모델명 Users -> User 수정

참조 모델 이름 수정 및 기타 모델명 정정

TeaCat-Develop committed:
💡 chore: makemigration

마이그레이션 초기화, 모델 추가 후 마이그레이션

TeaCat-Develop committed:
💡 chore: recruitment 마이그레이션파일 삭제 후 makemigration

마이그레이션 파일 충돌로 인해 다시 실행

TeaCat-Develop committed:
🎨 style: apps/recruitment/migrations/0001_initial.py code formatting

코드 포매팅

SWWS97 commented on apps/study_groups/models/group_member.py:
```diff
@@ -0,0 +1,21 @@
+    # 보호 필요 시: CASCADE-> PROTECT 변경
+    is_leader = models.BooleanField(default=False)
+    created_at = models.DateTimeField(auto_now_add=True)  # 처음 생성 시에만 자동 저장
+    updated_at = models.DateTimeField(auto_now=True)  # 갱신 시 저장
```
이부분도 apps/core/model.py에 있는 공통 추상화 모델 TimeStampedModel 참조해서 사용해주세요

TeaCat-Develop committed:
💡 chore: TimeStampModel 적용 및 마이그레이션

대상:
./apps/study_groups/models/group_member.py
./apps/study_groups/models/study_group.py

2joonkim reviewed (changes requested):
recruitment/migration 파일도 같이 들어왔습니다. 확인해보시고 마이그레이션 된 파일만 넣어주세요

SWWS97 committed:
Merge branch 'develop' into feat/study_group
